astroquery/casda/core.py

```diff
@@ -123,6 +141,253 @@ def _parse_result(self, response, verbose=False):
             print("Failed to convert query result to table", e)
             return response
 
+    def filter_out_unreleased(self, table):
+        """
+        Return a subset of the table which only includes released (public) data.
+
+        Parameters
+        ----------
+        table: `astropy.table.Table`
+            A table of results as returned by query_region. Must include an obs_release_date column.
+
+        Returns
+        -------
+        table : `astropy.table.Table`
+            The table with all unreleased (non public) data products filtered out.
+        """
+        now = str(datetime.now(timezone.utc).strftime('%Y-%m-%dT%H:%M:%S.%f'))
+        return table[(table['obs_release_date'] != '') & (table['obs_release_date'] < now)]
+
+    def stage_data(self, table, verbose=False):
+        """
+        Request access to a set of data files. All requests for data must use authentication. If you have access to the
+        data, the requested files will be brought online and a set of URLs to download the files will be returned.
+
+        Parameters
+        ----------
+        table: `astropy.table.Table`
+            A table describing the files to be staged, such as produced by query_region. It must include an
+            access_url column.
+        verbose: bool, optional
+            Should status message be logged periodically, defaults to False
+
+        Returns
+        -------
+        A list of urls of both the requested files and the checksums for the files
+        """
+        if not self._authenticated:
+            raise ValueError("Credentials must be supplied to download CASDA image data")
+
+        if table is None or len(table) == 0:
+            return []
+
+        # Use datalink to get authenticated access for each file
+        tokens = []
+        for row in table:
+            access_url = row['access_url']
+            response = self._request('GET', access_url, auth=self._auth,
+                                     timeout=self.TIMEOUT, cache=False)
+            response.raise_for_status()
+            soda_url, id_token = self._parse_datalink_for_service_and_id(response, 'cutout_service')
+            tokens.append(id_token)
+
+        # Create job to stage all files
+        job_url = self._create_soda_job(tokens, soda_url=soda_url)
+        if verbose:
+            print("Created data staging job", job_url)
```

Comment:
at some point we need to switch to use a proper logger rather than prints (see the usage of the astropy logger in e.g. the alma module). Currently the codebase is still a mixture of the two approaches, so it doesn't have to be done in this PR.